File: packages/camera/camera_android_camerax/android/src/test/java/io/flutter/plugins/camerax/ZoomStateTest.java
// Copyright 2013 The Flutter Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

package io.flutter.plugins.camerax;

import static org.junit.Assert.assertEquals;
import static org.mockito.Mockito.mock;
import static org.mockito.Mockito.when;

import androidx.camera.core.ZoomState;
import org.junit.Test;

public class ZoomStateTest {
  @Test
  public void minZoomRatio_returnsExpectedMinZoomRatio() {
    final PigeonApiZoomState api = new TestProxyApiRegistrar().getPigeonApiZoomState();

    final ZoomState instance = mock(ZoomState.class);
    final double value = 1.0;
    when(instance.getMinZoomRatio()).thenReturn((float) value);

    assertEquals(value, api.minZoomRatio(instance), 0.1);
  }

  @Test
  public void maxZoomRatio_returnsExpectedMaxZoomRatio() {
    final PigeonApiZoomState api = new TestProxyApiRegistrar().getPigeonApiZoomState();

    final ZoomState instance = mock(ZoomState.class);
    final double value = 1.0;
    when(instance.getMaxZoomRatio()).thenReturn((float) value);

    assertEquals(value, api.maxZoomRatio(instance), 0.1);
  }
}

File: packages/camera/camera_android_camerax/test/zoom_state_test.dart
// Copyright 2013 The Flutter Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

import 'package:camera_android_camerax/src/instance_manager.dart';
import 'package:camera_android_camerax/src/zoom_state.dart';
import 'package:flutter_test/flutter_test.dart';
import 'package:mockito/annotations.dart';

import 'test_camerax_library.g.dart';
import 'zoom_state_test.mocks.dart';

@GenerateMocks(<Type>[TestInstanceManagerHostApi])
void main() {
  TestWidgetsFlutterBinding.ensureInitialized();

  // Mocks the call to clear the native InstanceManager.
  TestInstanceManagerHostApi.setup(MockTestInstanceManagerHostApi());

  group('ZoomState', () {
    test('flutterApi create makes call to create expected ZoomState', () {
      final InstanceManager instanceManager = InstanceManager(
        onWeakReferenceRemoved: (_) {},
      );
      final ZoomStateFlutterApiImpl flutterApi = ZoomStateFlutterApiImpl(
        instanceManager: instanceManager,
      );
      const int zoomStateIdentifier = 68;
      const double minZoomRatio = 0;
      const double maxZoomRatio = 1;

      flutterApi.create(zoomStateIdentifier, minZoomRatio, maxZoomRatio);

      final ZoomState instance = instanceManager
          .getInstanceWithWeakReference(zoomStateIdentifier)! as ZoomState;
      expect(instance.minZoomRatio, equals(minZoomRatio));
      expect(instance.maxZoomRatio, equals(maxZoomRatio));
    });
  });
}
